### PR TITLE
Parse puppet 3 errors, add puppet 3 tests

### PR DESCRIPTION
**Pull request by rski**
This adds an error pattern for puppet 3 parser, so that it can parse some
errors which weren't parsed before. It also adds a test for this case.
Since puppet 3 and 4 parser output is incompatible, the puppet-parser
tests are now explicitly named after the puppet version they test.

### CODE REVIEW

File: test/flycheck-test.el [hunk 1]
```diff
@@ -3563,11 +3563,17 @@ Why not:
...
+; the puppet 4 and 3 tests are mutually exclusive due to one having column and the other not
+(flycheck-ert-def-checker-test puppet-parser puppet parser-error-puppet-4
   (flycheck-ert-should-syntax-check
    "language/puppet/parser-error.pp" 'puppet-mode
    '(3 9 error "Syntax error at '>'" :checker puppet-parser)))
 
+(flycheck-ert-def-checker-test puppet-parser puppet parser-error-puppet-3
+  (flycheck-ert-should-syntax-check
+   "language/puppet/parser-error2.pp" 'puppet-mode
+   '(4 nil error "Syntax error at 'helloagain'; expected '}'" :checker puppet-parser)))
```
Comment (lunaryorn) [inline test/flycheck-test.el]: Please skip either test when the corresponding puppet version is not available.  In other words the test for Puppet 3 errors shouldn't run with Puppet 4 and vice versa.

**Comment by lunaryorn** [comment]
@rski Please change the commit message to explain what issue is fixed, e.g. "Correctly parse puppet 3 errors" or similar.  Take a look at our [commit guidelines](http://www.flycheck.org/en/latest/contributor/contributing.html#commit-guidelines) :blush:

Furthermore please rebase on master.

**Comment by rski** [comment]
@lunaryorn the commit message and when-to-run-the-tests issues should be addressed now

**Comment by rski** [comment]
Gah, I ran into an edge case of this with some form of multi-line output from puppet-parser. Didn't know that was a thing. I'll close this pr until I get that working too.

**Comment by rski** [comment]
2016-08-07-122235_281x67_escrotum
It seems to work, I probably was testing this with the wrong version of flycheck.
@lunaryorn would you like to review/merge this?

**Comment by rski** [comment]
whhooooops I checked this with puppet 4, not 3.
